include/swift/ABI/Metadata.h

```diff
@@ -540,6 +540,14 @@ namespace {
 
 using TypeContextDescriptor = TargetTypeContextDescriptor<InProcess>;
 
+template<unsigned PointerSize>
+using ExternalTypeContextDescriptor
+#if SWIFT_OBJC_INTEROP
+= TargetTypeContextDescriptor<External<WithObjCInterop<RuntimeTarget<PointerSize>>>>;
```

Comment:
This still isn't quite right, since we want the client to be able to specify whether the target has ObjC interop separately from how the client builds. I believe the right approach here would be to modify `ProtocolConformanceDescriptorReader` to be templated on ObjC interop as well as pointer size, and do the `WithObjCInterop`/`NoObjCInterop` logic there. @adrian-prantl Does that sound right?

Reply:
Ah, makes perfect sense, thank you.
Updated with the type aliases templated on `ObjCInteropKind` to allow clients to decide on this separately from how this code is built.